### ISSUE TITLE
Frontend: upgrade to vue 3

### ISSUE THREAD

**Issue by stritti** [Sybit-Education/Diveni#219]
Currently we use vue.js v2 in combination with BoostrapVue. BootstrapVue does not support vue3.

* Check how to do migration to vue3 without loosing functionality.
* How could we solve lac of compatibility of BoostrapVue

See also #212

**Comment by stritti** [comment]
I think biggest issue will be BoostrapVue. There is already a fork for vue3 but not everything is supported currently: bootstrap-vue/bootstrap-vue#5196

Progress seems to be low on that within the project.

**Comment by stritti** [comment]
We should analys if this fork is worth to be used or if we should switch to another vue3 compatible UI framework.

**Comment by ravihidayat** [comment]
The freshest fork seems to not be maintained anymore. I am not sure about this, but I think we should consider talking about alternatives.

**Comment by stritti** [comment]
Yes, that is also my impression. Any suggestions for alternatives, @ravihidayat, @chrisWolter ?

**Comment by stritti** [comment]
If we switch to another ui framework, we have to verify that it supports

* vue 3
* vue-i18n
* responsive layout for mobile
* could be themed

Beside technical requirements we have to be as sure as possible that the ui framework is maintained in future and has a active community.

**Comment by stritti** [comment]
@ravihidayat  Many thanks for your proposals. IMHO PrimeVue would be a good base. The community is quite big. What would be your fav?

**Comment by ravihidayat** [comment]
@stritti I think I like PrimeVue more.

**Comment by stritti** [comment]
@ravihidayat then let us try to use PrimeVue together with vue 3.

**Comment by ravihidayat** [comment]
Let's. I'll start working on this and I'll keep you posted.

**Comment by ravihidayat** [comment]
@stritti Is there a way for us to discuss this since the discord link doesn't seem to work. I've managed to upgrade to Vue 3, and currently trying to debug. A bit challenging, so your input should be valuable.

**Comment by stritti** [comment]
> @stritti Is there a way for us to discuss this since the discord link doesn't seem to work. I've managed to upgrade to Vue 3, and currently trying to debug. A bit challenging, so your input should be valuable.

Discord don't work? Hm, I have to check it. Currently I am on a trip. I will be back on 12nd October. Then we could have a look together. 
What is your preferred timezone and time to talk?

Probably in time between @SponsoredByPuma or @KzuDemEvin could support?

**Comment by stritti** [comment]
@ravihidayat I found now the issue with discord. Sorry for that. Please try https://discord.gg/7JmRyn5dc6

**Comment by stritti** [comment]
@ravihidayat let me know how I could support you.

**Comment by ravihidayat** [comment]
@stritti Yeah, some of the bugs are persistent, and I'm not really sure what to do. I've even consulted to the docs. Should I DM you or should we talk over the development channel.

**Comment by stritti** [comment]
@ravihidayat I think easiest way could be to talk on dev channel. On next Monday, 17th I reserved the complete workday for Hacktoberfest and could be available between 10.00 am and 6.00 pm (GMT+2)

**Comment by stritti** [comment]
@ravihidayat Probably you could open a draft PR to have a discussion basis?

**Comment by stritti** [comment]
Hey @ravihidayat have you noticed last update of bootstrapVue: https://bootstrap-vue.org/vue3 ?

**Comment by haniAljti** [comment]
@stritti is anyone else working on this issue? because i could take if not.

**Comment by KzuDemEvin** [comment]
Hi @haniAljti, as far as I know no one else is working on this. The problem was that not all dependencies supported vue 3, so we decided to wait until all dependencies were updated. There were also compatibility issues with vue-bootstrap, but those may be fixed.
Feel free to work on it :)

**Comment by haniAljti** [comment]
Hi. Thanks for your answer. Ill try my best, please assign it to me

**Comment by haniAljti** [comment]
Hello,
Hope you're having a good day.

This is a list of the things that i did in my PR
- upgraded vue 2 to vue 3 using vue 3 compat
- updated and added a wrapper for Ui-toast-editor since the wrapper does not support vue 3 but it should function the same way.
- there is no update for "vue2-interact" that works on vue 3 but i've built the same functionality using Hammerjs. (I tested it. but it would be better if anyone tested it more)
- the project uses pinia instead of vuex now. Which should also solve #212
- updated other dependancies to use vue 3

Since there are alot of changes, please take your time reviewing.

**Comment by stritti** [comment]
Awesome great work. Was quite a big issue and now finally done. thanks all for your work!
